### PR TITLE
Add default function to AmazonS3FileSystem

### PR DESCRIPTION
**Pull request by matthewmturner**

### CODE REVIEW

**Comment by matthewmturner** [comment]
I dont think this worked well with implementing the `Default` trait because making a new `AmazonS3FileSystem` is async, so i just added a `default` function.  Let me know if any issues or if im missing something that i could have done with the `Default` trait.

File: src/object_store/aws.rs [hunk 1]
```diff
@@ -169,6 +169,13 @@ impl ObjectStore for AmazonS3FileSystem {
     }
 }
 
+#[allow(dead_code)]
+impl AmazonS3FileSystem {
+    async fn default() -> Self {
```
Comment (seddonm1) [inline src/object_store/aws.rs:174]: this needs to be `pub` right?

**Comment by matthewmturner** [comment]
@seddonm1 think we're good here?

**Comment by seddonm1** [comment]
@matthewmturner yes. merge it.

**Pull request merged**